### DISCUSSION TITLE
Removing robustness of the app

### DISCUSSION THREAD

**Discussion by exezbcz** [kodadot/nft-gallery#10370] [General]
More and more, I am thinking about removing the complexity and unused parts of the app. This discussion is made to choose the parts of apps that we will remove and comment out. 

## My picks
- create
- settings
- notifications
- explore card - remove networks etc. 
- explore? and make it more landing page exploration 
- on-chain identity
- featured articles - replace with some educative articles 
- modal flows - free mint, other complex stuff 

curious what you think

**Comment by JustLuuuu** [comment]
Add 
---
co-checker, explore generative art

remove
---
<img width="1672" alt="Screenshot 2024-05-29 at 14 44 33" src="https://github.com/kodadot/nft-gallery/assets/104380776/ac5e0c42-b9b1-40ae-969b-4b90f3b05fd6">

**Reply by prury** [reply]
hmmm, i kind of use it a lot

**Reply by vikiival** [reply]
Not very friendly tho.

**Reply by exezbcz** [reply]
so, as i view it, the exploring and the algorithm should be mainly on landing page.

**Comment by exezbcz** [comment]
@vikiival will comment on the most frontend-heavy features

**Comment by JustLuuuu** [comment]
> featured articles - replace with some educative articles 

Who will be writing those?

**Reply by exezbcz** [reply]
guess

**Comment by ThomasssArt** [comment]
> * create
> * settings
> * notifications
> * explore card - remove networks etc.
> * explore? and make it more landing page exploration
> * on-chain identity
> * featured articles - replace with some educative articles
> * modal flows - free mint, other complex stuff

create: as long as we dont want artists to create a collection by themself --> delete

settings: i dont see any useful benefit in it (but im not too deep into that) --> delete

notifications: no benefit --> delete

explore card - remove networks etc.: no benefit --> delete (what if we use a "top down" drop down instead of the actual?)
something like this:
![image](https://github.com/kodadot/nft-gallery/assets/60002234/2e0ee711-bab6-45d4-a284-6931ff641429)


on-chain identity: no benefit --> delete (i hope profiles coming soon :) )

featured articles - replace with some educative articles: maybe we can add a drop down for the "learn button" on the top and add these new educative articles + the actual stuff that is linked to the learn button there?! Than we could get rid of this whole section:
![image](https://github.com/kodadot/nft-gallery/assets/60002234/780a3fad-5490-4231-a516-dcf1e9e7e86d)

I would also get rid of these part (or redesign it):
![image](https://github.com/kodadot/nft-gallery/assets/60002234/c48de957-6cdf-4318-b2ee-27077ac9a2fd)


what about the cart function? is this needed/useful?
![image](https://github.com/kodadot/nft-gallery/assets/60002234/c7c908ec-ce32-4dc4-b17a-7f8fe9b6877d)

**Reply by ThomasssArt** [reply]
Just again regarding this section:
![image](https://github.com/kodadot/nft-gallery/assets/60002234/2bbf77b6-ea9d-4e1b-b89d-623136445cc4)

When you visit the koda website this is the first thing that a collector/artist/visitor or whatever sees. Imo it should be more eye catching and create some excitement. Thats why i would replace it (just my 2 cents).

**Reply by prury** [reply]
cart is very useful if you want to buy more than one NFT and its a really nice system

**Reply by exezbcz** [reply]
thanks for the insights, glad you agree with most of them

**Comment by prury** [comment]
things that i would remove: 

- migrate since we deactivate it and no longer support buying/minting on RMRK2(please confirm info with devs)
- Onchain ID as said before
- I would not remove notifications, but improve it to be off-chain(there is prob a issue for that)
- series-insight page
- user interface mode on settings
- metadata tab on settings
- generate waifu on ahk

**Comment by prury** [comment]
@exezbcz why would you remove create??

**Reply by vikiival** [reply]
> why would you remove create??

Have you seen anyone to create generative art except KodaMinter?

**Reply by prury** [reply]
true, true, exez enlightened me in dms

**Comment by vikiival** [comment]
> @vikiival will comment on the most frontend-heavy features


## Insane amount of fetches

I am not able to find the original issues but I have tried to load kodadot.xyz on uncached windows 10 PC (24 GB Ram, 8 core) with gigabit internet and this is what happened:

<img width="1914" alt="single-load" src="https://github.com/kodadot/nft-gallery/assets/22471030/65650d2b-644e-4031-afcf-cdc3ba8879d4">


Single page load toook **6 min and 30 sec** to finish.

That took 800 MB of data!
and more than **3000! Reqests**

Therefore would be optimal to find the source of this data overfetching :) as:
more request == more compute == higher load -> slow app


## Insane amount of requests / computation

- [Collection/activity](https://kodadot.xyz/ahp/collection/134/activity)

is fetching a big chunk of data to process therefore if user clicks there the app will freeze

Therefore data optimization -> a.k.a do graphql resolvers is a way to go.

## Others TODO

- On-chain identity should be ideally merged to profile (WDYT @exezbcz)
- Waifu minter should be hidden (as it is not used now)
- Simplify drop page

**Reply by preschian** [reply]
> Single page load toook **6 min and 30 sec** to finish.
> 
> That took 800 MB of data! and more than **3000! Reqests**

oohh, I think this is from service workers. from this plugin: https://github.com/kodadot/nft-gallery/blob/main/plugins/pwa.client.ts. let's remove then? 

SW will prefetch (non-blocking request) other assets in the background and put them in the cache, so the `finish` looks like it took longer to complete. Here is the page without cache and bypasses the service workers.

<img width="2672" alt="Screenshot 2024-06-06 at 22 50 51" src="https://github.com/kodadot/nft-gallery/assets/734428/265e907e-2200-4ae8-a2a5-3e2acc3e4e3e">

But we can still improve the number of requests.
- change the 301 status to render it directly. Currently, if the content is an image, it will redirect from image-workers to cf-images. It can reduce the requests by about 50 on that page
- Most of the requests are coming from the client's JS. Reducing the number of JS requests on the client can significantly improve the page load. But it's not an easy one. We need to revive this PR first: https://github.com/kodadot/nft-gallery/pull/9155. The thing is, the PR is not related to EVM. Should I revive that PR again?

**Reply by exezbcz** [reply]
ref: 
- https://github.com/kodadot/nft-gallery/issues/10416
